### DISCUSSION TITLE
A 2D mesh with circular region, and applying a constant force on the circle

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#23477] [Q&A Meshing] [answered]
Hello,

I am trying to analyze the problem in the image attached. Please how do I get a rectangular mesh with a circular region? And how do I apply a constant forcing term on the circular part? Thank you.

![image](https://user-images.githubusercontent.com/38023107/219877440-3ff9dd6e-e0ae-44aa-aef0-fa8a1ba5de17.png)

**Comment by GiudGiud** [comment]
Hello

For the constant forcing term, I think you will want to have a look at using a boundary condition if the variables only outside the circle in the rectangular regions.

To obtain a rectangular mesh with a circular hole, I think we have an example here:
https://github.com/idaholab/moose/blob/next/modules/navier_stokes/examples/flow-over-circle/mesh.i

Guillaume

**Reply by salaudeen-ya** [reply]
Thank you

**Comment by Rahim-Habibi** [comment, marked as answer]
Hi, 
I can recommed to use Gmsh for meshing the domian. 
 you can create rectangular mesh with a circular region in. 
If you want to apply boundary condition on the circle,  you can define the circle as a physical property then you will have access to that in MOOSE input file.

**Reply by salaudeen-ya** [reply]
Thank you!

**Comment by salaudeen-ya** [comment]
Using a circular body force specified over the region of interest also works
